src/agent/src/image_rpc.rs

```diff
@@ -138,7 +377,21 @@ impl ImageService {
 
         let source_creds = (!req.get_source_creds().is_empty()).then(|| req.get_source_creds());
 
-        Self::pull_image_from_registry(image, cid, &source_creds)?;
+        if Path::new(SKOPEO_PATH).exists() {
+            Self::pull_image_from_registry(image, cid, &source_creds)?;
+        } else {
+            let image = image.to_string();
+            let cid = cid.to_string();
+            let source_creds =
+                (!req.get_source_creds().is_empty()).then(|| req.get_source_creds().to_string());
+            tokio::task::spawn_blocking(move || {
```

Comment:
Any reason why we're spawning a thread for that? Why not calling `Self::pull_image_with_oci_distribution` directly? This way you can also reuse image, cid and source_creds directly.

Reply:
The reason we spawn a new thread is current `pull_image` is a async function, in `pull_image_with_oci_distribution`, we talk with attestation agent and download image will both do in async way, this will have "thread 'main' panicked at 'Cannot start a runtime from within a runtime." error.

Reply:
That's quite a shame. Could you please add a comment on that? It was not obvious to me at all.

Reply:
My bad, I should add the comment for that, added now :-)